signalfx-tracing/signalfx-go-tracing/gin/server/database/mgo.go

```diff
@@ -0,0 +1,80 @@
+package database
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/globalsign/mgo/bson"
+	"github.com/opentracing/opentracing-go"
+	mgotrace "github.com/signalfx/signalfx-go-tracing/contrib/globalsign/mgo"
+
+	"github.com/signalfx/tracing-examples/signalfx-tracing/signalfx-go-tracing/gin/server/models"
+)
+
+type mgoManager struct {
+	Host        string
+	Port        int
+	Name        string
+	ServiceName string
+}
+
+var _ Manager = (*mgoManager)(nil)
+
+var mgoSession *mgotrace.Session
+var parentSpan opentracing.Span
+
+// GetBoardByID returns a board for a given boardID
+func (m *mgoManager) GetBoardByID(c context.Context, id string) (models.Board, error) {
+	collection := m.getCollection(c)
+
+	board := models.Board{}
+	err := collection.Find(bson.M{"board_id": id}).One(&board)
+	if err != nil {
+		return models.Board{}, err
+	}
+	return board, nil
+}
+
+// InsertBoard inserts a given board
+func (m *mgoManager) InsertBoard(c context.Context, board models.Board) error {
+	collection := m.getCollection(c)
+
+	return collection.Insert(board)
+}
+
+// UpdateBoard saves a given updated board
+func (m *mgoManager) UpdateBoard(c context.Context, board models.Board) error {
+	collection := m.getCollection(c)
+
+	return collection.Update(bson.M{"board_id": board.ID}, board)
+}
+
+// getCollection returns board collection
+func (m *mgoManager) getCollection(c context.Context) *mgotrace.Collection {
+	if mgoSession == nil {
+		var ctx context.Context
+		parentSpan, ctx = opentracing.StartSpanFromContext(c, "mongo.session")
```

Comment:
Not sure it's a good idea to do it this way as an example for people since you'd never want to do this in a real app due to concurrent requests and the global variable. I guess you'd have to pass some kind of reference through the context they both can access if you want to do this. Or maybe just trace a single call like to `mgotrace.Dial`

Reply:
@jaysfx Just added another fix for it using `gin.Context.set/get` method as we discussed previously. Custom span looks right now.